code/modules/research/xenoarchaeology/tools/bunsen_burner.dm

```diff
@@ -5,56 +5,167 @@
+		var/obj/item/weapon/reagent_containers/R = W
 		if(held_container)
+			for(var/possible_fuel in possible_fuels)
+				if(R.reagents.has_reagent(possible_fuel) && (reagents.has_reagent(possible_fuel) || !reagents.reagent_list.len))
+					var/reagent_transfer = R.reagents.trans_id_to(src, possible_fuel, 10)
+					if(reagent_transfer)
+						to_chat(user, "<span class='notice'>You transfer [reagent_transfer]u of [possible_fuel] from \the [R] to \the [src]</span>")
+						return
 			to_chat(user, "<span class='warning'>You must remove the [held_container] first.</span>")
+			return
 		else
 			if(user.drop_item(W, src))
 				held_container = W
 				to_chat(user, "<span class='notice'>You put the [held_container] onto the [src].</span>")
 				var/image/I = image("icon"=W, "layer"=FLOAT_LAYER)
 				underlays += I
-				if(heating)
-					spawn(heat_time)
-						try_heating()
-
 				return 1 // avoid afterattack() being called
+	if(iswrench(W))
+		user.visible_message("<span class = 'warning'>[user] starts to deconstruct \the [src]!</span>","<span class = 'notice'>You start to deconstruct \the [src].</span>")
+		if(do_after(user, src, 5 SECONDS))
+			playsound(get_turf(src), 'sound/items/Ratchet.ogg', 50, 1)
+			drop_stack(/obj/item/stack/sheet/metal, loc, rand(3,4), user)
+			qdel(src)
+			return
 	else
-		to_chat(user, "<span class='warning'>You can't put the [W] onto the [src].</span>")
+		..()
+
+/obj/machinery/bunsen_burner/process()
+	if(held_container && heating)
+		var/turf/T = get_turf(src)
+		var/datum/gas_mixture/G = T.return_air()
+		if(!G || G.oxygen < 0.1)
+			visible_message("<span class = 'warning'>\The [src] splutters out from lack of oxygen.</span>","<span class = 'warning'>You hear something cough.</span>")
+			toggle()
+			return
+
+		var/max_temperature
+		var/thermal_energy_transfer
+		var/consumption_rate
+		var/unsafety = 0 //Possibility it lights things on its turf
+		var/o2_consumption
+		var/co2_consumption
+
+		for(var/possible_fuel in possible_fuels)
+			if(reagents.has_reagent(possible_fuel))
+				var/list/fuel_stats = possible_fuels[possible_fuel]
+				max_temperature = fuel_stats["max_temperature"]
+				thermal_energy_transfer = fuel_stats["thermal_energy_transfer"]
+				consumption_rate = fuel_stats["consumption_rate"]
+				unsafety = fuel_stats["unsafety"]
+				o2_consumption = fuel_stats["o2_cons"]
+				co2_consumption = fuel_stats["co2_cons"]
+
+				reagents.remove_reagent(possible_fuel, consumption_rate)
+				held_container.reagents.heating(thermal_energy_transfer, max_temperature)
+				G.adjust(o2 = -o2_consumption, co2 = -co2_consumption)
+				if(prob(unsafety) && T)
+					T.hotspot_expose(max_temperature, 5)
+				break
 
-/obj/machinery/bunsen_burner/attack_hand(mob/user as mob)
+		if(!max_temperature)
+			visible_message("<span class = 'warning'>\The [src] splutters out from lack of fuel.</span>","<span class = 'warning'>You hear something cough.</span>")
+			toggle()
+			return
+
+		/*
```

Comment:
Is there a reason to leave this comment block in?

Reply:
I was keeping it as reference to the old variables at the time, I'll get rid of them now.